### ISSUE TITLE
EJS Render Cache

### ISSUE THREAD

**Issue by zeim839** [ufosc/Club_Website_2#5]
TODO: Create a cache module for storing and asynchronously updating EJS parameters. 

We currently render dynamic pages by passing a JSON object like so:
```js
app.get('/', (req, res) => {
  res.render('index', { page: 'index' } /*  <=== Render parameters (JSON object)*/)
})
```
Once we get to putting together a club blog/announcements, etc., we'll have to create a database to pull the latest data (i.e latest blog posts), compile them into HTML elements, and then pass them as EJS render parameters as shown above. 

Instead of pinging the website and re-rendering for each request, we can set up a cache module. The cache module will include a global JSON variable which we pass to the render function like so:
```js
app.get('/', (req, res) => {
  res.render('index', cache.MY_COMPILED_PARAMS)
})
```
Meanwhile, the cache will run in the background as an asynchronous service, pinging the website once every X minutes, downloading new changes, compiling them into HTML and then updating the cache global variable - hence updating all our pages to the latest database content. The service doesn't need to run continuously either, we could program it to run AT MOST once every X minutes (instead of EVERY X minutes), this way we won't have to update the site if no one's reading it. 

As always, we could create a new `.env` parameter so users can configure their own update intervals.

The module should:
1. Pass render parameters via a JSON object (or maybe as the output of a function)
2. Be programmed to asynchronously update at most once every X minutes
3. Be written as an independent module, meaning:
    - It has a function for registering cache routines, i.e a function where you pass the name of the page and a callback function, and the callback function will output the value of cache[NAME OF PAGE] every X minutes.
4. Be configurable via a `config.js` / `.env`option. 

<b>NOTE: </b>Don't worry if this is confusing, if you're interested on working on this then I'll explain everything you'll need as we move forward. Just ask to be assigned and I'll reach out.

**Comment by amitanshusahu** [comment]
### not sure but i would like to work on this , may require help in the middle.. 😕

**Comment by zeim839** [comment]
Are you a member of the open source club? I think its best if we discuss this in person.

**Comment by amitanshusahu** [comment]
@zeim839 , what makes some one a member of open source club.. ? I m just here to contribute...

**Comment by willfpearce** [comment]
@zeim839 wouldn't we need to configure a database before this?

**Comment by zeim839** [comment]
Implemented in #29
